rir/src/api.cpp

```diff
@@ -222,8 +222,13 @@ SEXP pirCompile(SEXP what, const Assumptions& assumptions,
     logger.title("Compiling " + name);
     pir::Rir2PirCompiler cmp(m, logger);
     cmp.compileClosure(what, name, assumptions,
-                       [&](pir::ClosureVersion* c) {
+                       [&](pir::ClosureVersion* c, bool isNew) {
                            logger.flush();
+                           if (!isNew) {
+                               logger.debug("Reused already compiled version");
```

Comment:
if you put an `assert(false)` here, then we run through the whole test suite without triggering it. So this pr just adds dead code. we don't keep modules between compilations, so I am not sure how you think this could be triggered.